### PR TITLE
lib/runtime, dot/core, lib/babe: add transaction unit tests

### LINKED ISSUE #721: migrate runtime methods to runtime
## Expected Behavior

The runtime methods should all be in one location. The core and babe packages should not include runtime methods.

## Current Behavior

A few runtime methods are in the core and babe packages and should be with the other runtime methods in the runtime package.

### PR DESCRIPTION
**Pull request by noot**
## Changes

- add AsSignedExtrinsic to extrinsic.Transfer to convert Transfer to TransferExt (ie. with signature)
- add unit tests for various transaction types

## Tests:

```
go test ./... -short
```

## Checklist:

- [x] I have read [CONTRIBUTING](CONTRIBUTING.md) and [CODE_OF_CONDUCT](CODE_OF_CONDUCT.md)
- [ ] I have provided as much information as possible and necessary
- [ ] I have reviewed my own pull request before requesting a review
- [x] I have linked any relevant issues in my pull request comments
- [ ] All integration tests and required coverage checks are passing

## Issues:

related to #796 related to #810

### CODE REVIEW

File: lib/runtime/helpers_test.go [hunk 1]
```diff
@@ -192,3 +229,120 @@ func TestFinalizeBlock(t *testing.T) {
 		t.Fatalf("Fail: got %v expected %v", res, expected)
 	}
 }
+
+func TestApplyExtrinsic_IncludeData(t *testing.T) {
+	t.Skip()
+	// TOOD: this currently fails with a Bad Proof error.
```
Comment (noot) [inline lib/runtime/helpers_test.go:235]: need to investigate runtime, can do in #811

File: lib/runtime/helpers_test.go [hunk 2]
```diff
@@ -37,6 +41,39 @@ func TestValidateTransaction_IncludeData(t *testing.T) {
 	require.Equal(t, expected, validity)
 }
 
+func TestValidateTransaction_Transfer(t *testing.T) {
```
Comment (noot) [inline lib/runtime/helpers_test.go:44]: will add StorageChange and Authorities tests once #811 is done

File: lib/runtime/helpers_test.go [hunk 3]
```diff
@@ -192,3 +229,120 @@ func TestFinalizeBlock(t *testing.T) {
...
+	tx, err := ext.Encode()
+	require.NoError(t, err)
+
+	res, err := rt.ApplyExtrinsic(tx)
+	require.Nil(t, err)
+
+	require.Equal(t, []byte{0, 0}, res)
+}
+
+func TestApplyExtrinsic_Transfer_NoBalance(t *testing.T) {
```
Comment (noot) [inline lib/runtime/helpers_test.go:317]: will do other tests after #809  is figured out

File: lib/transaction/types.go [hunk 4]
```diff
@@ -61,5 +63,32 @@ func NewValidTransaction(extrinsic types.Extrinsic, validity *Validity) *ValidTr
...
+	binary.LittleEndian.PutUint64(buf, vt.Validity.Longevity)
+	enc = append(enc, buf...)
+
+	if vt.Validity.Propagate {
+		enc = append(enc, 1)
+	} else {
+		enc = append(enc, 0)
+	}
+
+	return enc, nil
```
Comment (noot) [inline lib/transaction/types.go:93]: idk why scale stopped working for this... will investigate

File: lib/runtime/helpers_test.go [hunk 5]
```diff
@@ -23,7 +27,7 @@ func TestValidateTransaction_IncludeData(t *testing.T) {
 	require.NoError(t, err)
 
 	validity, err := rt.ValidateTransaction(tx)
-	require.Nil(t, err)
+	require.NoError(t, err)
```
Comment (edwardmack) [inline lib/runtime/helpers_test.go:30]: I like this, I've been using require.Nil, I didn't even know about require.NoError, makes more sense.

**Comment by noot** [comment]
@edwardmack good point, I think only the Extrinsic is needed for apply_extrinsic, not the ValidTransaction.  I removed the ValidTransaction part from the `Set` test and it worked fine. (which makes sense since the encoding of the ValidTransaction has the extrinsic portion first)

**Pull request merged**